src/plugins/proxy/src/plugin.cpp

```diff
@@ -378,11 +393,25 @@ std::shared_ptr<ov::IRemoteContext> ov::proxy::Plugin::get_default_context(const
 
 std::shared_ptr<ov::ICompiledModel> ov::proxy::Plugin::import_model(std::istream& model,
                                                                     const ov::AnyMap& properties) const {
-    auto dev_name = get_fallback_device(get_device_from_config(properties));
-    auto device_config = construct_device_config(dev_name, m_configs, properties);
+    auto dev_name = get_device_name();
+    auto dev_idx = get_device_from_config(properties);
+    auto has_dev_idx = is_device_in_config(properties);
+    auto is_new_api = get_core()->is_new_api();
+
+    auto device = get_fallback_device(get_device_from_config(properties));
+    auto device_config = construct_device_config(device, m_configs, properties);
+    auto device_model = get_core()->import_model(model, device, device_config);
+    ov::SoPtr<ov::IRemoteContext> device_context;
+    ov::RemoteContext remote_context;
+    try {
+        device_context = {device_model->get_context(), device_model._so};
+        remote_context = ov::proxy::RemoteContext::make_context(ov::SoPtr<ov::IRemoteContext>(
+            std::make_shared<ov::proxy::RemoteContext>(device_context, dev_name, dev_idx, has_dev_idx, is_new_api),
+            device_model._so));
+    } catch (const ov::NotImplemented&) {
+    }
```

Comment:
It seems code is duplicated in 2 places. Maybe create function for `RemoteContext` creation?